### PR TITLE
Responsive UI / Redesign

### PR DESCRIPTION
**Pull request by dbaines**
### General/UI

* Massive UI redesign and responsive considerations 
* Infinitely nested Koi::Menu.items with new menu  
* You can create pinned links at the top of the interface by adding Koi::PriorityMenu.items to your koi.rb file  
* Themable CSS file (`assets/stylesheets/koi/_overrides.scss`)
* Using the browser back button after clicking on tabs will now activate the appropriate tab  

### Keyboard Shortcuts

* alt + m = toggle menu, focus on first link 
* esc = close menu, close lightbox

### Model Index

* Clicking on an image will now show it in a lightbox rather than a link to the image in a new tab 

### Model Show

* Redesigned default show page 

### Model Edit/New

* Users are now warned when there are unsaved changes on the page if they attempt to leave the page. 

### Asset Manager

* Now in-page full screen
* Drag and drop file uploader
* Moved filters and search in to a modal 
* Tagging now uses select2 like the rest of Koi rather than it's own tagging implementation 

### Sitemap

* Added touch support 
* Exposed keys to the sitemap list with an icon that the user can hover revealing what the key is 
* Added icon for hidden nodes 
* Added locking of sitemap dragging 
* Removed "God" menu from navitem modals since they don't work with caching 
* Removed duplicate "Url" field from navitem models 

### Crud Config

* New `:latlng` form field type - draggable pin on a map to save lat/lng values, added to `super_hero` for testing 

### Developer UI Styleguide 

* Available at `/admin/styleguide` and available in the menu when logged in as a superadmin  

### Issues Solved

#122 - Remove "Add to Page" in asset manager when not in CKEditor
#125 - Fullscreen asset manager
#160,#176 - CKEditor issues
#179 - Lockable sitemap 
#183 - Map form field
#202 - Pagination for asset manager
#211 - Inline Nested script ordering
#219 - Asset manager callbacks for editing after uploading 
#224 - Merged search and tag filters in asset manager 
#226 - Deleting an asset wouldn't ask for confirmation 
#227 - Removed NavItem show for mobile

### CODE REVIEW

**Comment by Obversity** [comment]
> Users are now warned when there are unsaved changes on the page if they attempt to leave the page.

I notice this doesn't affect going back in the browser. Would it even be possible to do for that?

**Comment by Obversity** [comment]
> Users are now warned when there are unsaved changes on the page if they attempt to leave the page.

I notice this doesn't affect going back in the browser if you go straight from typing to the back button—presumably because the field's on-change event hasn't fired? Anyhow, cool feature!

**Comment by Obversity** [comment]
@dbaines  The new latlng field type is awesome. 😄

**Comment by Obversity** [comment]
@dbaines The sitemap looks great, and feels much more usable—and much safer. 

Would it be possible to remove the 'drag' cursor when hovering over sitemap ietms when the sitemap is 'drag locked'?

File: app/assets/javascripts/koi/koi/datePicker.js [hunk 1]
```diff
@@ -1,23 +1,23 @@
-! function ($)
-{
```
Comment (dbaines) [inline app/assets/javascripts/koi/koi/datePicker.js]: I think this might be a false-positive in Github. The file no longer exists in the project.

**Comment by dbaines** [comment]
@Obversity I've pushed up a fix for sitemap cursor still being 'move' when the sitemap is locked.

**Pull request merged**